sklearn/feature_selection/tests/test_from_model.py

```diff
@@ -607,3 +607,63 @@ def importance_getter(estimator):
         warnings.simplefilter("error", UserWarning)
 
         selector.transform(X.iloc[1:3])
+
+
+@pytest.mark.parametrize(
+    "error, err_msg, max_features",
+    (
+        [ValueError, "max_features == 10, must be <= 4", 10],
+        [TypeError, "'max_features' must be either an int or a callable", "a"],
+        [ValueError, r"max_features\(X\) == 5, must be <= 4", lambda x: x.shape[1] + 1],
+    ),
+)
+def test_partial_fit_validate_max_features(error, err_msg, max_features):
+    """Test that partial_fit from SelectFromModel validates `max_features`."""
+    X, y = datasets.make_classification(
+        n_samples=100,
+        n_features=4,
+        random_state=0,
+    )
+
+    with pytest.raises(error, match=err_msg):
+        SelectFromModel(
+            estimator=SGDClassifier(), max_features=max_features
+        ).partial_fit(X, y, classes=[0, 1])
+
+
+def test_partial_fit_validate_feature_names():
```

Comment:
Thanks @thomasjpfan. After a bit of investigation, I found that currently `test_pandas_column_name_consistency` test doesn't cover `partial_fit` method in `SelectFromModel`, because  `SelectFromModel` is tested using `LogisticRegression(C=1)` which has no `partial_fit` method.  

I am thinking if we could test `SelectFromModel` using `SGDRegressor` instead in `test_common.py`.  What do you think?
```python
def _construct_instance(Estimator):
    """Construct Estimator instance if possible."""
    required_parameters = getattr(Estimator, "_required_parameters", [])
    if len(required_parameters):
        if required_parameters in (["estimator"], ["base_estimator"]):
            # `RANSACRegressor` will raise an error with any model other
            # than `LinearRegression` if we don't fix `min_samples` parameter.
            # For common test, we can enforce using `LinearRegression` that
            # is the default estimator in `RANSACRegressor` instead of `Ridge`.
            if issubclass(Estimator, RANSACRegressor):
                estimator = Estimator(LinearRegression())
            elif issubclass(Estimator, RegressorMixin):
                estimator = Estimator(Ridge())
            elif issubclass(Estimator, SelectFromModel):
                estimator = Estimator(SGDRegressor())
            else:
                estimator = Estimator(LogisticRegression(C=1))
```

Reply:
Good point. The common test failed because of the randomness of `SGDRegressor`. A quick fix is:

```python
elif issubclass(Estimator, SelectFromModel):
    # Increases coverage because SGDRegressor has partial_fit
    estimator = Estimator(SGDRegressor(random_state=0))
```

I still think the current `test_partial_fit_validate_feature_names` is good to keep because it covers the case where `estimator_` is a classifier.

Reply:
Thanks again. I have updated the common test while keeping the current `test_partial_fit_validate_feature_names`.